### ISSUE TITLE
doesn't build on Scala 2.12.0

### ISSUE THREAD

**Issue by SethTisue** [scala/scala-abide#74]
after

``` diff
-    scalaVersion := "2.11.8",
+    scalaVersion := "2.12.0-RC2",
```

then

```
> abide/test
[info] Compiling 16 Scala sources to /Users/tisue/scala-abide/abide/target/scala-2.12.0-RC2/classes...
[trace] Stack trace suppressed: run last abide/compile:compileIncremental for the full output.
[error] (abide/compile:compileIncremental) scala.reflect.internal.Types$TypeError: type mismatch;
[error]  found   : List[scala.tools.abide.Rule{val context(): scala.tools.abide.Context{val universe(): AbidePlugin.this.global.type}}]
[error]  required: List[scala.tools.abide.Rule{val context(): scala.tools.abide.Context{val universe(): AbidePlugin.this.global.type}}]
[error] Total time: 7 s, completed Oct 26, 2016 3:28:27 PM
> last abide/compile:compileIncremental
[...]
scala.reflect.internal.Types$TypeError: type mismatch;
 found   : List[scala.tools.abide.Rule{val context(): scala.tools.abide.Context{val universe(): AbidePlugin.this.global.type}}]
 required: List[scala.tools.abide.Rule{val context(): scala.tools.abide.Context{val universe(): AbidePlugin.this.global.type}}]

    at scala.tools.nsc.typechecker.Contexts$ThrowingReporter.handleError(Contexts.scala:1374)
    at scala.tools.nsc.typechecker.Contexts$ContextReporter.issue(Contexts.scala:1226)
    at scala.tools.nsc.typechecker.Contexts$Context.issue(Contexts.scala:575)
    at scala.tools.nsc.typechecker.ContextErrors$ErrorUtils$.issueTypeError(ContextErrors.scala:106)
    at scala.tools.nsc.typechecker.ContextErrors$ErrorUtils$.issueNormalTypeError(ContextErrors.scala:99)
    at scala.tools.nsc.typechecker.ContextErrors$TyperContextErrors$TyperErrorGen$.AdaptTypeError(ContextErrors.scala:219)
    at scala.tools.nsc.typechecker.Typers$Typer.adaptMismatchedSkolems$1(Typers.scala:1044)
    at scala.tools.nsc.typechecker.Typers$Typer.lastTry$1(Typers.scala:1055)
    at scala.tools.nsc.typechecker.Typers$Typer.adaptExprNotFunMode$1(Typers.scala:1105)
    at scala.tools.nsc.typechecker.Typers$Typer.vanillaAdapt$1(Typers.scala:1151)
    at scala.tools.nsc.typechecker.Typers$Typer.adapt(Typers.scala:1195)
    at scala.tools.nsc.typechecker.Typers$Typer.runTyper$1(Typers.scala:5514)
    at scala.tools.nsc.typechecker.Typers$Typer.typedInternal(Typers.scala:5530)
    at scala.tools.nsc.typechecker.Typers$Typer.body$2(Typers.scala:5474)
    at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5478)
    at scala.tools.nsc.typechecker.Typers$Typer.typedByValueExpr(Typers.scala:5559)
    at scala.tools.nsc.typechecker.Typers$Typer.typedAssign$1(Typers.scala:4321)
    at scala.tools.nsc.typechecker.Typers$Typer.typedOutsidePatternMode$1(Typers.scala:5428)
    at scala.tools.nsc.typechecker.Typers$Typer.typedInAnyMode$1(Typers.scala:5457)
    at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5464)
    at scala.tools.nsc.typechecker.Typers$Typer.runTyper$1(Typers.scala:5500)
    at scala.tools.nsc.typechecker.Typers$Typer.typedInternal(Typers.scala:5530)
    at scala.tools.nsc.typechecker.Typers$Typer.body$2(Typers.scala:5474)
    at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5478)
    at scala.tools.nsc.typechecker.Typers$Typer.typedByValueExpr(Typers.scala:5559)
    at scala.tools.nsc.typechecker.Typers$Typer.typedStat$1(Typers.scala:3058)
    at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$8(Typers.scala:3187)
    at scala.tools.nsc.typechecker.Typers$Typer.typedStats(Typers.scala:3187)
    at scala.tools.nsc.typechecker.Typers$Typer.typedBlock(Typers.scala:2439)
    at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typed1$91(Typers.scala:5422)
    at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typerWithLocalContext$1(Typers.scala:490)
    at scala.tools.nsc.typechecker.Typers$Typer.typedOutsidePatternMode$1(Typers.scala:490)
    at scala.tools.nsc.typechecker.Typers$Typer.typedInAnyMode$1(Typers.scala:5457)
    at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5464)
    at scala.tools.nsc.typechecker.Typers$Typer.runTyper$1(Typers.scala:5500)
    at scala.tools.nsc.typechecker.Typers$Typer.typedInternal(Typers.scala:5530)
    at scala.tools.nsc.typechecker.Typers$Typer.body$2(Typers.scala:5474)
    at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5478)
    at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5570)
    at scala.tools.nsc.typechecker.Typers$Typer.typedIf$1(Typers.scala:4338)
    at scala.tools.nsc.typechecker.Typers$Typer.typedOutsidePatternMode$1(Typers.scala:5423)
    at scala.tools.nsc.typechecker.Typers$Typer.typedInAnyMode$1(Typers.scala:5457)
    at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5464)
    at scala.tools.nsc.typechecker.Typers$Typer.runTyper$1(Typers.scala:5500)
    at scala.tools.nsc.typechecker.Typers$Typer.typedInternal(Typers.scala:5530)
    at scala.tools.nsc.typechecker.Typers$Typer.body$2(Typers.scala:5474)
    at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5478)
    at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedArg$1(Typers.scala:3200)
    at scala.tools.nsc.typechecker.Typers$Typer.typedArg(Typers.scala:481)
    at scala.tools.nsc.typechecker.Typers$Typer.typedArgToPoly$1(Typers.scala:3590)
    at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$doTypedApply$33(Typers.scala:3598)
    at scala.tools.nsc.typechecker.Typers$Typer.handlePolymorphicCall$1(Typers.scala:3598)
    at scala.tools.nsc.typechecker.Typers$Typer.doTypedApply(Typers.scala:3609)
    at scala.tools.nsc.typechecker.Typers$Typer.normalTypedApply$1(Typers.scala:4647)
    at scala.tools.nsc.typechecker.Typers$Typer.typedApply$1(Typers.scala:4681)
    at scala.tools.nsc.typechecker.Typers$Typer.typedInAnyMode$1(Typers.scala:5447)
    at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5464)
    at scala.tools.nsc.typechecker.Typers$Typer.runTyper$1(Typers.scala:5500)
    at scala.tools.nsc.typechecker.Typers$Typer.typedInternal(Typers.scala:5530)
    at scala.tools.nsc.typechecker.Typers$Typer.body$2(Typers.scala:5474)
    at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5478)
    at scala.tools.nsc.typechecker.Typers$Typer.typedByValueExpr(Typers.scala:5559)
    at scala.tools.nsc.typechecker.Typers$Typer.typedStat$1(Typers.scala:3058)
    at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$8(Typers.scala:3187)
    at scala.tools.nsc.typechecker.Typers$Typer.typedStats(Typers.scala:3187)
    at scala.tools.nsc.typechecker.Typers$Typer.typedBlock(Typers.scala:2439)
    at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typed1$91(Typers.scala:5422)
    at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typerWithLocalContext$1(Typers.scala:490)
    at scala.tools.nsc.typechecker.Typers$Typer.typedOutsidePatternMode$1(Typers.scala:490)
    at scala.tools.nsc.typechecker.Typers$Typer.typedInAnyMode$1(Typers.scala:5457)
    at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5464)
    at scala.tools.nsc.typechecker.Typers$Typer.runTyper$1(Typers.scala:5500)
    at scala.tools.nsc.typechecker.Typers$Typer.typedInternal(Typers.scala:5530)
    at scala.tools.nsc.typechecker.Typers$Typer.body$2(Typers.scala:5474)
    at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5478)
    at scala.tools.nsc.typechecker.Typers$Typer.transformedOrTyped(Typers.scala:5709)
    at scala.tools.nsc.typechecker.Typers$Typer.typedDefDef(Typers.scala:2270)
    at scala.tools.nsc.typechecker.Typers$Typer.typedMemberDef$1(Typers.scala:5412)
    at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5463)
    at scala.tools.nsc.typechecker.Typers$Typer.runTyper$1(Typers.scala:5500)
    at scala.tools.nsc.typechecker.Typers$Typer.typedInternal(Typers.scala:5530)
    at scala.tools.nsc.typechecker.Typers$Typer.body$2(Typers.scala:5474)
    at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5478)
    at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5555)
    at scala.tools.nsc.typechecker.Typers$Typer.typedPos(Typers.scala:5562)
    at scala.tools.nsc.transform.Fields$FieldsTransformer.typedPos(Fields.scala:514)
    at scala.tools.nsc.transform.AccessorSynthesis$CheckedAccessorTreeSynthesis$SynthLazyAccessorsIn.$anonfun$expandLazyClassMember$2(AccessorSynthesis.scala:347)
    at scala.tools.nsc.transform.AccessorSynthesis$CheckedAccessorTreeSynthesis$SynthLazyAccessorsIn.$anonfun$expandLazyClassMember$1(AccessorSynthesis.scala:347)
    at scala.tools.nsc.transform.AccessorSynthesis$CheckedAccessorTreeSynthesis$SynthLazyAccessorsIn.expandLazyClassMember(AccessorSynthesis.scala:347)
    at scala.tools.nsc.transform.Fields$FieldsTransformer.transform(Fields.scala:718)
    at scala.tools.nsc.transform.Fields$FieldsTransformer.transformTermsAtExprOwner(Fields.scala:745)
    at scala.tools.nsc.transform.Fields$FieldsTransformer.$anonfun$transformStats$3(Fields.scala:758)
    at scala.tools.nsc.transform.Fields$FieldsTransformer.transformStats(Fields.scala:758)
    at scala.tools.nsc.transform.Fields$FieldsTransformer.transformStats(Fields.scala:513)
    at scala.reflect.internal.Trees.itransform(Trees.scala:1408)
    at scala.reflect.internal.Trees.itransform$(Trees.scala:1340)
    at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:16)
    at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:16)
    at scala.reflect.api.Trees$Transformer.transform(Trees.scala:2555)
    at scala.tools.nsc.transform.TypingTransformers$TypingTransformer.super$transform(TypingTransformers.scala:40)
    at scala.tools.nsc.transform.TypingTransformers$TypingTransformer.$anonfun$transform$1(TypingTransformers.scala:40)
    at scala.tools.nsc.transform.TypingTransformers$TypingTransformer.atOwner(TypingTransformers.scala:30)
    at scala.tools.nsc.transform.TypingTransformers$TypingTransformer.transform(TypingTransformers.scala:25)
    at scala.tools.nsc.transform.Fields$FieldsTransformer.transform(Fields.scala:737)
    at scala.tools.nsc.transform.Fields$FieldsTransformer.transform(Fields.scala:513)
    at scala.reflect.api.Trees$Transformer.transformTemplate(Trees.scala:2563)
    at scala.reflect.internal.Trees.$anonfun$itransform$4(Trees.scala:1412)
    at scala.tools.nsc.transform.TypingTransformers$TypingTransformer.atOwner(TypingTransformers.scala:30)
    at scala.tools.nsc.transform.TypingTransformers$TypingTransformer.atOwner(TypingTransformers.scala:25)
    at scala.reflect.internal.Trees.itransform(Trees.scala:1411)
    at scala.reflect.internal.Trees.itransform$(Trees.scala:1340)
    at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:16)
    at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:16)
    at scala.reflect.api.Trees$Transformer.transform(Trees.scala:2555)
    at scala.tools.nsc.transform.TypingTransformers$TypingTransformer.transform(TypingTransformers.scala:44)
    at scala.tools.nsc.transform.Fields$FieldsTransformer.transform(Fields.scala:737)
    at scala.tools.nsc.transform.Fields$FieldsTransformer.$anonfun$transformStats$4(Fields.scala:758)
    at scala.tools.nsc.transform.Fields$FieldsTransformer.transformStats(Fields.scala:758)
    at scala.tools.nsc.transform.Fields$FieldsTransformer.transformStats(Fields.scala:513)
    at scala.reflect.internal.Trees.$anonfun$itransform$7(Trees.scala:1430)
    at scala.tools.nsc.transform.TypingTransformers$TypingTransformer.atOwner(TypingTransformers.scala:30)
    at scala.tools.nsc.transform.TypingTransformers$TypingTransformer.atOwner(TypingTransformers.scala:25)
    at scala.reflect.internal.Trees.itransform(Trees.scala:1430)
    at scala.reflect.internal.Trees.itransform$(Trees.scala:1340)
    at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:16)
    at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:16)
    at scala.reflect.api.Trees$Transformer.transform(Trees.scala:2555)
    at scala.tools.nsc.transform.TypingTransformers$TypingTransformer.super$transform(TypingTransformers.scala:40)
    at scala.tools.nsc.transform.TypingTransformers$TypingTransformer.$anonfun$transform$2(TypingTransformers.scala:42)
    at scala.tools.nsc.transform.TypingTransformers$TypingTransformer.atOwner(TypingTransformers.scala:30)
    at scala.tools.nsc.transform.TypingTransformers$TypingTransformer.transform(TypingTransformers.scala:25)
    at scala.tools.nsc.transform.Fields$FieldsTransformer.transform(Fields.scala:737)
    at scala.tools.nsc.transform.Fields$FieldsTransformer.transform(Fields.scala:513)
    at scala.tools.nsc.ast.Trees$Transformer.transformUnit(Trees.scala:140)
    at scala.tools.nsc.transform.Transform$Phase.apply(Transform.scala:30)
    at scala.tools.nsc.Global$GlobalPhase.$anonfun$applyPhase$1(Global.scala:402)
    at scala.tools.nsc.Global$GlobalPhase.applyPhase(Global.scala:395)
    at scala.tools.nsc.Global$GlobalPhase.$anonfun$run$1(Global.scala:366)
    at scala.tools.nsc.Global$GlobalPhase.$anonfun$run$1$adapted(Global.scala:366)
    at scala.collection.AbstractIterator.foreach(Iterator.scala:932)
    at scala.tools.nsc.Global$GlobalPhase.run(Global.scala:366)
    at scala.tools.nsc.Global$Run.compileUnitsInternal(Global.scala:1404)
    at scala.tools.nsc.Global$Run.compileUnits(Global.scala:1389)
    at scala.tools.nsc.Global$Run.compileSources(Global.scala:1384)
    at scala.tools.nsc.Global$Run.compile(Global.scala:1478)
    at xsbt.CachedCompiler0.run(CompilerInterface.scala:116)
    at xsbt.CachedCompiler0.run(CompilerInterface.scala:95)
    at xsbt.CompilerInterface.run(CompilerInterface.scala:26)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at sbt.compiler.AnalyzingCompiler.call(AnalyzingCompiler.scala:107)
    at sbt.compiler.AnalyzingCompiler.compile(AnalyzingCompiler.scala:53)
    at sbt.compiler.AnalyzingCompiler.compile(AnalyzingCompiler.scala:47)
    at sbt.compiler.MixedAnalyzingCompiler$$anonfun$compileScala$1$1.apply$mcV$sp(MixedAnalyzingCompiler.scala:50)
    at sbt.compiler.MixedAnalyzingCompiler$$anonfun$compileScala$1$1.apply(MixedAnalyzingCompiler.scala:50)
    at sbt.compiler.MixedAnalyzingCompiler$$anonfun$compileScala$1$1.apply(MixedAnalyzingCompiler.scala:50)
    at sbt.compiler.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:74)
    at sbt.compiler.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:49)
    at sbt.compiler.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:64)
    at sbt.compiler.IC$$anonfun$compileInternal$1.apply(IncrementalCompiler.scala:160)
    at sbt.compiler.IC$$anonfun$compileInternal$1.apply(IncrementalCompiler.scala:160)
    at sbt.inc.IncrementalCompile$$anonfun$doCompile$1.apply(Compile.scala:66)
    at sbt.inc.IncrementalCompile$$anonfun$doCompile$1.apply(Compile.scala:64)
    at sbt.inc.IncrementalCommon.cycle(IncrementalCommon.scala:32)
    at sbt.inc.Incremental$$anonfun$1.apply(Incremental.scala:72)
    at sbt.inc.Incremental$$anonfun$1.apply(Incremental.scala:71)
    at sbt.inc.Incremental$.manageClassfiles(Incremental.scala:99)
    at sbt.inc.Incremental$.compile(Incremental.scala:71)
    at sbt.inc.IncrementalCompile$.apply(Compile.scala:54)
    at sbt.compiler.IC$.compileInternal(IncrementalCompiler.scala:160)
    at sbt.compiler.IC$.incrementalCompile(IncrementalCompiler.scala:138)
    at sbt.Compiler$.compile(Compiler.scala:155)
    at sbt.Compiler$.compile(Compiler.scala:141)
    at sbt.Defaults$.sbt$Defaults$$compileIncrementalTaskImpl(Defaults.scala:879)
    at sbt.Defaults$$anonfun$compileIncrementalTask$1.apply(Defaults.scala:870)
    at sbt.Defaults$$anonfun$compileIncrementalTask$1.apply(Defaults.scala:868)
    at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
    at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
    at sbt.std.Transform$$anon$4.work(System.scala:63)
    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
    at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
    at sbt.Execute.work(Execute.scala:237)
    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
    at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
    at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
[error] (abide/compile:compileIncremental) scala.reflect.internal.Types$TypeError: type mismatch;
[error]  found   : List[scala.tools.abide.Rule{val context(): scala.tools.abide.Context{val universe(): AbidePlugin.this.global.type}}]
[error]  required: List[scala.tools.abide.Rule{val context(): scala.tools.abide.Context{val universe(): AbidePlugin.this.global.type}}]
> 
```

**Comment by SethTisue** [comment]
once this is addressed, Abide could be added to the 2.12 community build:

```diff
+  scala-abide-uri:              "https://github.com/scala/scala-abide.git"

+  ${vars.base} {
+    name: "scala-abide"
+    uri:  ${vars.uris.scala-abide-uri}
+    extra.exclude: ["root"]  // dbuild thinks it tries to publish root#root
+  }
```

**Comment by SethTisue** [comment]
same error with 2.12.0 (perhaps needless to say)

**Comment by SethTisue** [comment]
fixed by #79

**Comment by SethTisue** [comment]
community build: https://github.com/scala/community-builds/pull/397/files
